runhouse/servers/http/http_server.py

```diff
@@ -185,6 +196,57 @@ def get_object(message: Message):
             media_type="application/json",
         )
 
+    @staticmethod
+    @app.get("/run_object")
+    def get_run_object(message: Message):
+        from runhouse import Run
+        from runhouse.rns.api_utils.utils import resolve_absolute_path
+
+        HTTPServer.register_activity()
+        logger.info(
+            f"Message received from client to get run object: {b64_unpickle(message.data)}"
+        )
+        run_name, folder_path = b64_unpickle(message.data)
+
+        folder_path = folder_path or Run._base_cluster_folder_path(run_name)
+        folder_path_on_system = resolve_absolute_path(folder_path)
+
+        try:
+            HTTPServer.register_activity()
```

Comment:
We don't need this again so soon after calling it above, only if a reasonable amount of time elapses.